openmoltools/amber.py

```diff
@@ -223,7 +253,13 @@ def run_antechamber(molecule_name, input_filename, charge_method="bcc", net_char
     if frcmod_filename is None:
         frcmod_filename = molecule_name + '.frcmod'
 
-    cmd = "antechamber -i %s -fi mol2 -o %s -fo mol2 -s 2" % (input_filename, gaff_mol2_filename)
+    #Use temporary directory to do this to avoid issues with spaces in filenames, etc.
+    tempdir = tempfile.mkdtemp()
+    startdir = os.getcwd()
+    shutil.copy( input_filename, os.path.join( tempdir, 'in.mol2') )
+    os.chdir( tempdir )
```

Comment:
Can you explain a little more or give a code snippet? I'm not understanding what you mean here. Maybe I also don't know the context manager thing you are mentioning.



—
Sent from Mailbox

On Thu, Jun 25, 2015 at 7:42 PM, Kyle Beauchamp <notifications@github.com>
wrote:

>> @@ -223,7 +253,13 @@ def run_antechamber(molecule_name, input_filename, charge_method="bcc", net_char
>>      if frcmod_filename is None:
>>          frcmod_filename = molecule_name + '.frcmod'
>>  
>> -    cmd = "antechamber -i %s -fi mol2 -o %s -fo mol2 -s 2" % (input_filename, gaff_mol2_filename)
>> +    #Use temporary directory to do this to avoid issues with spaces in filenames, etc.
>> +    tempdir = tempfile.mkdtemp()
>> +    startdir = os.getcwd()
>> +    shutil.copy( input_filename, os.path.join( tempdir, 'in.mol2') )
>> +    os.chdir( tempdir )
> Do we want to be resistant to users getting errors (e.g. control-C) during the directory switch?  If so, it could be useful to implement this using the context manager protocol, which automatically handles bad things happening inside the block.  Up to you.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/choderalab/openmoltools/pull/138/files#r33324894